canonical/microk8s
# Use API server proxy from cluster-agent and remove traefik

#### Summary


Remove traefik from the snap, use the API server proxy from https://github.com/canonical/microk8s-cluster-agent/pull/7

#### Changes


- Add a hook that runs during the `configure` hook, migrates the old traefik configs into the new format expected by the apiserver proxy command
- Remove traefik service, add apiserver-proxy
- Change dqlite worker node joins to setup properly

#### Testing


Tested the following scenarios:

1. Join worker node into an existing cluster, ensure it can connect properly and spin up new pods.
2. Join worker node into an existing cluster with multiple nodes. Ensure it automatically picks up all API servers.
3. Join worker node with traefik proxy, upgrade to new snap, ensure configuration is migrated properly and that kubelet comes up.
4. Add new control plane to cluster after worker joins, ensure the API server proxy picks it up and uses it.
5. Remove control plane node from cluster, ensure worker drops it from the configuration.

#### Possible Regressions


#### Notes

PR on the cluster-agent repository: https://github.com/canonical/microk8s-cluster-agent/pull/7
